File: dayTwo/pTwo.js
const fs = require('fs');

var depth = 0;
var x = 0;
var aim = 0;
var instruct = [];

fs.readFile('input', (err, data) => {
	if (err) throw err;
	instruct = data.toString().split('\n');
	instruct.pop();
	for (var i = 0; i < instruct.length; i++) {
		if (instruct[i].startsWith("forward")) {
            console.log("forth");
			x += parseInt(instruct[i].split(" ")[1]);
            depth += aim*parseInt(instruct[i].split(" ")[1]);
		} else if (instruct[i].startsWith("up")) {
            console.log("aim upth");
			aim -= parseInt(instruct[i].split(" ")[1]);
		} else if (instruct[i].startsWith("down")) {
            console.log("aim downth");
			aim += parseInt(instruct[i].split(" ")[1]);
		}
		console.log("X: ", x, ", Depth: ", depth, ", Aim: ", aim);
	}
	console.log("Multiplied: ", x*depth);
});

File: dayTwo/cpp/pTwo.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

vector<string> instruct;
int depth = 0;
int x = 0;
int aim = 0;

int main() {
	ifstream in;
	in.open("../../input");
	if (in.is_open()) {
		while(in.good()) {
			string line;
			while(getline(in, line)) {
				cout << line << endl;
				instruct.push_back(line);
			}
		}
	}
	for (int i=0; i < instruct.size(); i++) {
		size_t pos = 0;
		if (instruct.at(i).rfind("forward", 0) == 0) {
			pos = instruct.at(i).find(" ");
			int num = stoi(instruct.at(i).substr(pos, 2));
			x += num;
			depth += aim*num;
		} else if (instruct.at(i).rfind("up", 0) == 0) {
			pos = instruct.at(i).find(" ");
			aim -= stoi(instruct.at(i).substr(pos, 2));
		} else if (instruct.at(i).rfind("down", 0) == 0) {
			pos = instruct.at(i).find(" ");
			aim += stoi(instruct.at(i).substr(pos, 2));
		}
		cout << "X: " << x << ", Depth: " << depth << ", Aim: " << aim << endl;
	}

	cout << "Multiplied: " << x*depth << endl; 

	return 0;
}

